src/robot_interface/models/mission/task.py

```diff
@@ -68,7 +69,8 @@ class DriveToPose(MotionTask):
     """
 
     pose: Pose
-    task_name: Literal["drive_to_pose"] = "drive_to_pose"
+    name: Literal["drive_to_pose"] = "drive_to_pose"
+    id: UUID = field(default_factory=uuid4)
```

Comment:
Could the id attribute be a part of the Task dataclass, since all subclasses needs it?

Reply:
This would be preferrable, however dataclasses in python have a weird limitation here.

If you define a field with a default value in a parent dataclass the inheriting dataclass cannot have any fields without default values. This is because the inheritance for dataclasses "stacks" fields in such a way that the default field then comes before the non-default field.

Reply:
I think it would be solved by using regular classes where the behavior might be a bit different if I remember correctly. Although that's not really a solution.

The same limitation is why `tag_id` is present in both `TakeImage` and `TakeThermalImage` but not in `InspectionTask`.

Reply:
I agree with @aestene here. I suggest that we keep this implementation for now, but looking at ways to improve the dataclass implmentation is left for issue #133.